volatility3/framework/plugins/windows/pe_symbols.py

```diff
@@ -0,0 +1,777 @@
+            except exceptions.VolatilityException:
+                continue
+
+            # this is not expected - it means pdbconv broke when parsing the PDB
+            except TypeError as e:
+                vollog.error(
+                    f"Unable to parse PDB file for module {pdb_name} -> {e}. Please file a bug on the GitHub issue tracker."
+                )
+
+        # cannot do anything without the symbols
+        if not mod_symbols:
+            return None
+
+        pdb_module = context.module(
+            mod_symbols, layer_name=layer_name, offset=module_start
+        )
+
+        return PDBSymbolFinder(layer_name, mod_name, module_start, pdb_module)
+
+    @staticmethod
+    def _find_symbols_through_pdb(
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        module_instances: List[Tuple[str, int, int]],
+        mod_name: str,
+    ) -> Generator[PDBSymbolFinder, None, None]:
+        """
+        Attempts to resolve the symbols in `wanted_symbols` through PDB analysis
+        """
+        for module_info in module_instances:
+            mod_module = PESymbols._get_pdb_module(
+                context, config_path, mod_name, module_info
+            )
+            if mod_module:
+                yield mod_module
+
+    @staticmethod
+    def _find_symbols_through_exports(
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        module_instances: List[Tuple[str, int, int]],
+        mod_name: str,
+    ) -> Generator[ExportSymbolFinder, None, None]:
+        """
+        Attempts to resolve the symbols in `wanted_symbols` through export analysis
+        """
+        pe_table_name = intermed.IntermediateSymbolTable.create(
+            context, config_path, "windows", "pe", class_types=pe.class_types
+        )
+
+        # for each process layer and VAD, construct a PE and examine the export table
+        for module_info in module_instances:
+            exported_symbols = PESymbols._get_exported_symbols(
+                context, pe_table_name, mod_name, module_info
+            )
+            if exported_symbols:
+                yield exported_symbols
+
+    @staticmethod
+    def _get_symbol_value(
+        wanted_modules: PESymbolFinder.cached_value_dict,
+        mod_name: str,
+        symbol_resolver: PESymbolFinder,
+    ) -> Generator[Tuple[str, int], None, None]:
+        """
+        Enumerates the symbols specified as wanted by the calling plugin
+        """
+        wanted_symbols = wanted_modules[mod_name]
+
+        if (
+            PESymbols.wanted_names not in wanted_symbols
+            and PESymbols.wanted_addresses not in wanted_symbols
+        ):
+            vollog.warning(
+                f"Invalid `wanted_symbols` sent to `find_symbols` for module {mod_name}. addresses and names keys both misssing."
+            )
+            return
+
+        symbol_keys = [
```

Comment:
It seems like you understood this later.

Reply:
I did eventually, but not because the variable was names well.  Can we make it more descriptive please?  Something that will help is gonna be putting the actual methods in, rather than the strings and later doing a getattr...